app/controllers/concerns/inherited_proofing_session.rb

```diff
@@ -0,0 +1,27 @@
+module InheritedProofingSession
+  extend ActiveSupport::Concern
+  include InheritedProofingConcern
+
+  included do
+    before_action :destroy_inherited_proofing, if: -> { inherited_proofing? }
+  end
+
+  private
+
+  def destroy_inherited_proofing
+    inherited_proofing_clear_session
+    inherited_proofing_log_analytics
+  end
+
+  def inherited_proofing_clear_session
+    user_session['idv/inherited_proofing'] = {}
```

Comment:
what if we added this line to the existing controller and that was it?

Reply:
Hi @zachmargolis I _could_ have done this, I assume you mean _unconditionally_ clear out the IP session regardless of whether or not IP is in play; however, we can't likewise unconditionally log analytics for IP - that requires a conditional statement, otherwise our analytics would be inaccurate. I _could_ have also added a conditional statement to log analytics for IP as well, but I wanted to avoid cluttering up the existing `SessionsController` with conditional statements, albeit (for now) just the two of them. I believe isolating the IP code in the `InheritedProofingSession` concern and conditionally cleaning up IP session and logging analytics 1) provides a logical sense of single responsibility 2) keeps things physically/logically segregated without complicating things. The `before_action` callbacks work well here, and keep things uncluttered; I think the aforementioned benefits are worth the cost of a little indirection IMHO.

Reply:
Hi @zachmargolis,

> what if we added this line to the existing controller and that was it?

If you can be more specific as to what exactly you don't like about this, I can try to address it.

Reply:
so my thinking is, "how can we keep the whole deletion simple?" as in, minimal changes, minimal lines of code, minimal new things to understand. It looks like the base sessions controller already deletes `idv/in_person` and `idv/doc_auth` implying that's the only places those things store session data. So my thought is, if we added a similar "delete user_session['idv/inherited_proofing']` could we replace the entire mixin with just a single line?

Reply:
Hi @zachmargolis,
I addressed why there was more to simply clearing out `user_session['idv/inherited_proofing']` in the `SessionController` 2 comments back. I moved some code from the `InheritedProofingSession` concern to the `SessionsController`, and done away with the `before_action` callbacks. The code _is_ simple. The `SessionsController` code  has been cleaned up and is now logically refactored into members, whereas before, it was just dumped into the #destroy action. The IP code is tucked away in its own concern; easy to view, easy to deal with. I hope that suffices because we're supposed to demo this at 4PM EST.